### PR TITLE
Don't show particles when splash events are canceled

### LINKED ISSUE #8626: WaterBottleSplashEvent & PotionSplashEvent still show particle effects when canceled
### Expected behavior

Canceling the event cancels particle effects

### Observed/Actual behavior

Particle effects are not canceled

### Steps/models to reproduce

Throw splash potion and try to cancel event

### Plugin and Datapack List

none

### Paper version

latest 1.19.3 dev build

### Other

_No response_

### PR DESCRIPTION
**Pull request by Lulu13022002**
Closes #8626 
This also cancel the glass sound
and include the lingering event to avoid inconsistencies

### CODE REVIEW

**Comment by kennytv** [comment]
This should probably also go into the `0648-Fix-PotionSplashEvent-for-water-splash-potions` patch, changing its description to encompass the different issues it fixes with the event/adding the new event

File: patches/server/0946-Don-t-show-particles-when-splash-event-are-canceled.patch [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
++                    showParticles = this.makeAreaOfEffectCloud(itemstack, potionregistry); // Paper
+                 } else {
+-                    this.applySplash(list, hitResult != null && hitResult.getType() == HitResult.Type.ENTITY ? ((EntityHitResult) hitResult).getEntity() : null); // Paper - nullable hitResult
++                    showParticles = this.applySplash(list, hitResult != null && hitResult.getType() == HitResult.Type.ENTITY ? ((EntityHitResult) hitResult).getEntity() : null); // Paper - nullable hitResult / cancel particle when event is canceled
+                 }
+             }
+ 
+             int i = potionregistry.hasInstantEffects() ? 2007 : 2002;
+ 
++            if (showParticles) // Paper
```
Comment (Machine-Maker) [inline patches/server/0946-Don-t-show-particles-when-splash-event-are-canceled.patch]: Use brackets to improve readability and increase chance of conflict if something changes.

File: patches/server/0946-Don-t-show-particles-when-splash-event-are-canceled.patch [hunk 2]
```diff
@@ -0,0 +1,77 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Lulu13022002 <41980282+Lulu13022002@users.noreply.github.com>
+Date: Fri, 9 Dec 2022 13:08:16 +0100
+Subject: [PATCH] Don't show particles when splash event are canceled
```
Comment (Machine-Maker) [inline patches/server/0946-Don-t-show-particles-when-splash-event-are-canceled.patch]: Put in exiting patch as kenny said.

**Review by Machine-Maker — APPROVED** [review]

File: patches/server/0946-Don-t-show-particles-when-splash-event-are-canceled.patch (hunk 2 above)
Comment (Lulu13022002) [reply patches/server/0946-Don-t-show-particles-when-splash-event-are-canceled.patch]: Done as mentioned

**Review by kennytv — APPROVED** [review]

**Pull request merged**